WMF Framework/Event Platform/StorageManager.swift

```diff
@@ -0,0 +1,220 @@
+                    }
+                    events.append(PersistedEvent(data: record.data, stream: stream, managedObjectURI: record.objectID.uriRepresentation()))
+                    count += 1
+                }
+                DDLogDebug("EPC: Found \(count) events awaiting submission")
+            } catch let error {
+                DDLogError(error.localizedDescription)
+            }
+        }
+        return events
+    }
+
+    func markPosted(event: PersistedEvent, date: Date) {
+        perform { moc in
+            do {
+                guard let psc = moc.persistentStoreCoordinator else {
+                    DDLogWarn("EPC: Error getting persistent store coordinator")
+                    return
+                }
+                guard let moid = psc.managedObjectID(forURIRepresentation: event.managedObjectURI) else {
+                    DDLogWarn("EPC: Error getting managed object ID for URI \(event.managedObjectURI)")
+                    return
+                }
+                guard let record = try moc.existingObject(with: moid) as? EPEventRecord else {
+                    DDLogWarn("EPC: Tried to mark managed object \(moid) as posted, but it was not found.")
+                    return
+                }
+                record.posted = date
+                self.save(moc)
+            } catch let error {
+                DDLogError(error.localizedDescription)
+            }
+        }
+    }
+
+    func markFailed(event: PersistedEvent) {
+        perform { moc in
+            do {
+                guard let psc = moc.persistentStoreCoordinator else {
+                    DDLogWarn("EPC: Error getting persistent store coordinator")
+                    return
+                }
+                guard let moid = psc.managedObjectID(forURIRepresentation: event.managedObjectURI) else {
+                    DDLogWarn("EPC: Error getting managed object ID for URI \(event.managedObjectURI)")
+                    return
+                }
+                guard let record = try moc.existingObject(with: moid) as? EPEventRecord else {
+                    DDLogWarn("EPC: Tried to mark managed object \(moid) as failed, but it was not found.")
+                    return
+                }
+                record.failed = true
+                self.save(moc)
+            } catch let error {
+                DDLogError(error.localizedDescription)
+            }
+
+        }
+    }
+
+    func pruneStaleEvents(completion: @escaping (() -> Void)) {
+        perform { moc in
+            defer {
+                completion()
+            }
+
+            let pruneFetch = NSFetchRequest<NSFetchRequestResult>(entityName: "WMFEPEventRecord")
+            pruneFetch.returnsObjectsAsFaults = false
+
+            let pruneDate = Date().addingTimeInterval(-(self.pruningAge)) as NSDate
+            pruneFetch.predicate = NSPredicate(format: "(recorded < %@) OR (posted != nil) OR (failed == TRUE)", pruneDate)
```

Comment:
@mdholloway @tonisevener Are we actually using the distinction between `posted` and `failed` anywhere? Functionally speaking, both of them mean "not eligible for POSTing and need to be pruned". It seems like we could just have a single property called `pruneable` or something. We can replace `mark{Posted,Failed}` with `markPruneable`.

Reply:
+1, again I was mimicking the older behavior, but I'm all for consolidating this unless there's some reason not to.  @tonisevener thoughts?

Reply:
@mdholloway @linehan yep agreed, I'm not aware of any need for the distinction and a single flag sounds way better. Nice cleanup catch!

Reply:
OK, I'm still going to hold off on re-pushing while I work on updating the input buffer stuff, but for the record, this is done.